app/register-organization/page.tsx

```diff
@@ -0,0 +1,283 @@
+"use client";
+import React, { useState } from "react";
+import { NextPage } from "next";
+import { Button } from "@/components/ui/button";
+import AuthenticationPageTemplate from "@/components/authentication-page-template";
+import { Input } from "@/components/ui/input";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+  FormLabel,
+} from "@/components/ui/form";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from "@/components/ui/command";
+
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+import { Check, ChevronDown } from "lucide-react";
+import { cn } from "@/lib/utils";
+
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { ORG_SIZE_OPTIONS, COUNTRIES, TIMEZONES } from "@/utils/constants";
+import { organizationDefaultValues } from "@/utils/forms/initial-values";
+import { OrganizationFormDetail } from "@/utils/forms/interfaces";
+import { organizationSchema } from "@/utils/forms/validations";
+import { useCreateOrganizationMutation } from "@/redux/apis/auth-api";
+import { toast } from "sonner";
+import { useRouter } from "next/navigation";
+import { ROUTES } from "@/utils/constants";
+
+const SignUp: NextPage = () => {
+  const router = useRouter();
+  const [countryPopover, setCountryPopover] = useState<boolean>(false);
+  const [timeZonePopover, setTimeZonePopover] = useState<boolean>(false);
+
+  const [createOrganization, { isLoading }] = useCreateOrganizationMutation();
+
+  const form = useForm({
+    resolver: yupResolver(organizationSchema),
+    defaultValues: organizationDefaultValues,
+  });
+
+  const onSubmit = async (formValues: OrganizationFormDetail) => {
+    try {
+      const res = await createOrganization(formValues).unwrap();
+      if (res?.data) {
+        toast.success(res?.message);
+        form.reset();
+        setTimeout(() => {
+          router.replace(ROUTES.DASHBOARD);
+        }, 500);
+      }
+    } catch (err: any) {
+      toast.error(err?.data?.message);
+    }
+  };
+
+  return (
+    <AuthenticationPageTemplate className="w-full">
+      <Form {...form}>
+        <form
+          onSubmit={form.handleSubmit(onSubmit)}
+          className="flex flex-col gap-6"
+        >
+          <FormField
+            control={form.control}
+            name={"name"}
+            render={({ field }) => (
+              <FormItem>
+                <FormLabel required>Organization name</FormLabel>
```

Comment:
use the form data approach like the ones used in sign in and sign up e.g., email.label